File: AuraLogbook.Api/Models/MoodEntry.cs
using AuraLogbook.Api.Models.Enums;

namespace AuraLogbook.Api.Models;
public class MoodEntry
{
    public int Id { get; set; }
    public int UserId { get; set; }
    public DateOnly Date { get; set; }
    public List<MoodType> Moods { get; set; } = new();
    public string? Comment { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

File: aura-ui/src/features/mood/models/MoodEntry.ts
import type { MoodType } from "./MoodType";

/**
 * Represents a mood entry submitted by a user.
 */
export interface MoodEntry {
  id: number;
  userId: number;
  date: string; // ISO string format (e.g., "2025-06-21") for DateOnly
  moods: MoodType[];
  comment?: string;
  createdAt: string; // ISO timestamp (e.g., "2025-06-21T14:32:00Z")
}
